tools/validation/findings.py

```
@@ -0,0 +1,76 @@
+    confidence: float
+    rationale: str
+    suggested_fix: str | None = None
+
+
+@runtime_checkable
+class OracleLike(Protocol):
+    domain: str
+    column_map: dict[str, str]
+    thresholds: dict[str, float]
+
+    def reference_frame(self, dataset: str, keys: pl.DataFrame) -> pl.DataFrame | None: ...
```

Flake8 (7.3.0)

[error] multiple statements on one line (def)

(E704)